code/modules/reagents/chemistry/reagents/reaction_agents_reagents.dm

```diff
@@ -0,0 +1,113 @@
+    holder.remove_reagent(type, volume)
+    return TRUE
+    
+/datum/reagent/reaction_agent/basic_buffer
+    name = "Strong basic buffer"
+    description = "This reagent will consume itself and move the pH of a beaker towards alkalinity when added to another."
+    color = "#3853a4"
+    ph = 14
+    ///The strength of the buffer where (volume/holder.total_volume)*strength. So for 1u added to 50u the ph will increase by 0.4
+    var/strength = 20
+
+/datum/reagent/reaction_agent/basic_buffer/intercept_reagents_transfer(datum/reagents/target, amount)
+    . = ..()
+    if(. == FALSE)
+        return FALSE
+    if(target.ph >= ph)
+        target.my_atom.audible_message("<span class='warning'>The beaker froths as the buffer is added, to no effect.</span>")
+        playsound(target.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
+        holder.remove_reagent(type, amount)//Remove from holder because it's not transfered
+        return TRUE 
+    var/ph_change = (amount/target.total_volume)*strength
+    target.adjust_all_reagents_ph(ph_change, 0, ph)
+    target.my_atom.audible_message("<span class='warning'>The beaker froths as the ph changes!</span>")
+    playsound(target.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
+    holder.remove_reagent(type, amount)
+    return TRUE
+    
+/datum/reagent/reaction_agent/purity_tester
+    name = "Purity tester"
+    description = "This reagent will consume itself and violently react if there is a highly impure reagent in the beaker."
+    ph = 3
+    color = "#3853a4"
+
+/datum/reagent/reaction_agent/purity_tester/intercept_reagents_transfer(datum/reagents/target, amount)
+    . = ..()
+    if(. == FALSE) //Surely there's a better way to do this
+        return FALSE
+    var/is_inverse = FALSE
+    for(var/_reagent in target.reagent_list)
+        var/datum/reagent/reaction_agent/reagent = _reagent
+        if(reagent.purity <= reagent.inverse_chem_val)
+            is_inverse = TRUE
+    if(is_inverse)
+        target.my_atom.audible_message("<span class='warning'>The beaker bubbles violently as the reagent is added!</span>")
+        playsound(target.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
+    else
+        target.my_atom.audible_message("<span class='warning'>The added reagent doesn't seem to do much.</span>")
+    holder.remove_reagent(type, amount)
+    return TRUE
+
+/datum/reagent/reaction_agent/speed_agent
+    name = "Reaction speed agent"
+    description = "This reagent will consume itself and speed up an ongoing reaction, modifying the current reaction's purity by it's own."
+    ///How much the reaction speed is sped up by - for 5u added to 100u, an additional step of 0.5 will be done
+    var/strength = 10
+
+/datum/reagent/reaction_agent/speed_agent/intercept_reagents_transfer(datum/reagents/target, amount)
+    . = ..()
+    if(. == FALSE)
+        return FALSE
+    if(!length(holder.reaction_list))//Safetly check and div/0 avoid
+        return FALSE
```

Comment:
Is this expected to happen? If not, put a `stack_trace()`

Reply:
Yes, this can happen normally, you can add this reagent to a beaker with no ongoing reactions. I felt it was too mean to consume the reagent if there was nothing to speed up.